network-api/networkapi/wagtailpages/pagemodels/buyersguide/products.py

```diff
@@ -246,15 +247,28 @@ def get_votes(self):
         votes = [int(x) for x in self.vote_bins.split(",")]
         return votes
 
-    def get_most_voted(self):
+    def get_vote_average(self):
         votes = self.get_votes()
         vote_breakdown = {k: v for (k, v) in enumerate(votes)}
-        highest = max(vote_breakdown, key=vote_breakdown.get)
-        label = self.get_vote_labels()[highest]
+        total_votes = sum(vote_breakdown.values())
+        """If there's no votes, let the user know (also protects from division by zero errors)"""
+        if total_votes == 0:
+            label = self.get_vote_labels()[5]
+            return {
+                "bin": 0,
+                "value": 0,
+                "label": label[0],
+                "localized": label[1]
+            }
+        weighted_votes = 0
+        for i in range(len(vote_breakdown)):
+            weighted_votes += vote_breakdown[i] * i
+        average_vote = round(weighted_votes / total_votes)
+        label = self.get_vote_labels()[average_vote]
```

Comment:
I'm low on blood sugar atm so I can't really do math properly. Apologies if I'm adding more confusion here 😅 Should the index be `[average_vote-1]`?

Reply:
It originally was, but for some reason everything got shifted up one creepiness level. I'm investigating this.

Reply:
I remember why. `vote_breakdown`` starts at 0, so it's already adjusted for the array. No adjustment needed here unless I'm missing something else (and I could be lol).